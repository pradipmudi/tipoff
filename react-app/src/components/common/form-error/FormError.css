.tipoff-notification {
    margin-bottom:10px;
}