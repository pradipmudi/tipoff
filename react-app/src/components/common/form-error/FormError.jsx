import React from 'react';
import {Notification} from 'bloomer';

import './FormError.css';

export default function FormError({hideError,errors}) {
    return (
        <div>
            <Notification isHidden={hideError} isColor="danger" className="tipoff-notification">
                {errors && errors.length ? (
                    <ul>
                        {errors.map((err,index) => 
                                <li key={index}>{err}</li>
                        )}
                    </ul>
                ) : (
                    <span></span>
                )}
            </Notification>
        </div>
    )
};
