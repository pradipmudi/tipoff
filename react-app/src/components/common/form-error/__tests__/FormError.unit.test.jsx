import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import FormError from '../FormError';

describe("FormError", () => {
    describe("has errors", () => {
        let props, wrapper;
        beforeAll(() => {
            props = {
                hideError : false,
                errors : [`Error 1 occured`, `Error 2 occurred`]
            }
            wrapper = shallow(<FormError {...props} />);
        });

        it("matches snapshot", () => {
            const snap = renderer.create(<FormError {...props} />).toJSON();
            expect(snap).toMatchSnapshot();
        });

        it("renders without issues", () => {
            expect(wrapper.exists()).toEqual(true);
        });

        it("renders Notification component", () => {
            expect(wrapper.find('withHelpersModifiers(Notification)').length).toEqual(1);
        });

        it("renders same number of li tags as number of errors", () => {
            expect(wrapper.find('withHelpersModifiers(Notification) li').length).toEqual(props.errors.length);
        });

        it("renders error message as passed in the props", () => {
            expect(wrapper.find('withHelpersModifiers(Notification) li').first().text()).toEqual(props.errors[0]);
        });

        it("renders isHidden property same as props", () => {
            expect(wrapper.find('withHelpersModifiers(Notification)').prop('isHidden')).toEqual(props.hideError);
        });

        it("renders danger color for notification", () => {
            expect(wrapper.find('withHelpersModifiers(Notification)').prop('isColor')).toEqual('danger');
        });
    })

    describe("doesn't have errors", () => {
        let props, wrapper;
        beforeAll(() => {
            props = {
                hideError : true,
                errors : []
            }
            wrapper = shallow(<FormError {...props} />);
        });

        it("matches snapshot", () => {
            const snap = renderer.create(<FormError {...props} />).toJSON();
            expect(snap).toMatchSnapshot();
        });

        it("renders without issues", () => {
            expect(wrapper.exists()).toEqual(true);
        });

        it("renders Notification component", () => {
            expect(wrapper.find('withHelpersModifiers(Notification)').length).toEqual(1);
        });

        it("doesn't render ul ", () => {
            expect(wrapper.find('withHelpersModifiers(Notification) ul').length).toEqual(0);
        });

        it("renders isHidden property same as props", () => {
            expect(wrapper.find('withHelpersModifiers(Notification)').prop('isHidden')).toEqual(props.hideError);
        });
    })
    
});