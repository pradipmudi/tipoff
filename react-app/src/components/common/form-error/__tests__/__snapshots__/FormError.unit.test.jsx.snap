// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormError doesn't have errors matches snapshot 1`] = `
<div>
  <div
    className="notification is-danger is-hidden tipoff-notification"
  >
    <span />
  </div>
</div>
`;

exports[`FormError has errors matches snapshot 1`] = `
<div>
  <div
    className="notification is-danger tipoff-notification"
  >
    <ul>
      <li>
        Error 1 occured
      </li>
      <li>
        Error 2 occurred
      </li>
    </ul>
  </div>
</div>
`;
